client/src/components/body/landingpage/citation/ProfileGeneration.js

```diff
@@ -66,11 +67,39 @@ function initSlideShow(slideshow) {
 } 
 
 class ProfileGeneration extends Component{
+    constructor(props) {
+      super(props);
+      this.state = {teamInformation: [],professorInformation:[]};
+    }
+
+    setTeamInformation=(data)=>{
+      this.setState({teamInformation: data})
+    }
+    setProfessorInformation=(data)=>{
+      this.setState({professorInformation: data})
+    }
+
     componentDidMount() {
+        if(this.state.professorInformation.length===0){
+          async function fetchProfessors() {
+              return TeamMemberService.getAllProfessors();
+          }
+          fetchProfessors().then((data) => {
+            this.setProfessorInformation(data);
+          });
+        }
+        if(this.state.teamInformation.length===0){
+          async function fetchTeam() {
+              return TeamMemberService.getAllTeamMembers();
+          }
+          fetchTeam().then((data) => {
+            this.setTeamInformation(data);
+            slideshows = document.querySelectorAll('[data-component="slideshow"]');
+            slideshows.forEach(initSlideShow);
```

Comment:
orEach returns undefined most times so it might be better to use .map() instead as it doesn't return undefined and it doesn't mutate the array